tests/template_tests/syntax_tests/test_basic.py

```diff
@@ -311,3 +311,15 @@ def test_basic_syntax38(self):
         """
         output = self.engine.render_to_string('basic-syntax38', {"var": {"callable": lambda: "foo bar"}})
         self.assertEqual(output, 'foo bar')
+
+    @setup({'template': '{% block content %}'})
+    def test_unclosed_block(self):
+        msg = "Unclosed tag 'block'. Looking for one of: endblock."
+        with self.assertRaisesMessage(TemplateSyntaxError, msg):
+            self.engine.render_to_string('template')
+
+    @setup({'template': '{% if a %}'})
+    def test_unclosed_block2(self):
+        msg = "Unclosed tag 'if'. Looking for one of: elif, else, endif."
+        with self.assertRaisesMessage(TemplateSyntaxError, msg):
+            self.engine.render_to_string('template')
```

Comment:
These tests were added because the detailed unclosed block tag exception from the debug lexer is now the default. I couldn't find tests for that elsewhere.